tests/unit-tests/test_dataloading.py

```diff
@@ -77,10 +83,474 @@
+                                        "C_13": 5,
+                                        "C_4": 1,
+                                        "C_16": 0,
+                                        "C_7": 2
+                                    }
+                                },
+                                "transformation_name": "DistCategoryTransformation"
+                            }
+                        }
+                    ],
+                    "labels": [
+                        {
+                            "column": "",
+                            "type": "link_prediction",
+                            "split_rate": {
+                                "train": 0.8,
+                                "val": 0.1,
+                                "test": 0.1
+                            }
+                        }
+                    ]
+                },
+                {
+                    "data": {
+                        "format": "parquet",
+                        "files": [
+                            "edges/paper_is-about_subject.parquet"
+                        ]
+                    },
+                    "source": {
+                        "column": "source_id",
+                        "type": "paper"
+                    },
+                    "dest": {
+                        "column": "dest_id",
+                        "type": "subject"
+                    },
+                    "relation": {
+                        "type": "is-about"
+                    }
+                },
+                {
+                    "data": {
+                        "format": "parquet",
+                        "files": [
+                            "edges/paper_written-by_author.parquet"
+                        ]
+                    },
+                    "source": {
+                        "column": "source_id",
+                        "type": "paper"
+                    },
+                    "dest": {
+                        "column": "dest_id",
+                        "type": "author"
+                    },
+                    "relation": {
+                        "type": "written-by"
+                    }
+                },
+                {
+                    "data": {
+                        "format": "parquet",
+                        "files": [
+                            "edges/subject_has_paper.parquet"
+                        ]
+                    },
+                    "source": {
+                        "column": "source_id",
+                        "type": "subject"
+                    },
+                    "dest": {
+                        "column": "dest_id",
+                        "type": "paper"
+                    },
+                    "relation": {
+                        "type": "has"
+                    }
+                }
+            ],
+            "is_homogeneous": "false"
+        },
+        "version": "gsprocessing-v1.0"
```

Comment:
You may also need a homogeneous graph config.

Reply:
added.